features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/presequite/MemoryRequirementChecker.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.presequite
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.core.internal.persistence.file.canReadSafe
+import com.datadog.android.core.internal.persistence.file.existsSafe
+import com.datadog.android.core.internal.persistence.file.readLinesSafe
+import java.io.File
+
+internal class MemoryRequirementChecker(
+    private val memInfoFile: File = File(MEM_INFO_PATH),
+    private val internalLogger: InternalLogger
+) : SystemRequirementChecker {
+
+    private var checkedValue: Long? = null
+    override fun checkMinimumRequirement(systemRequirementConfiguration: SystemRequirementConfiguration): Boolean {
+        return getMaxHeapSize() >= systemRequirementConfiguration.memorySizeInMb
+    }
+
+    override fun name(): String = MEMORY_CHECK_NAME
+
+    override fun checkedValue(): Any? = checkedValue
+
+    private fun getMaxHeapSize(): Long {
+        if (!(memInfoFile.existsSafe(internalLogger) && memInfoFile.canReadSafe(internalLogger))) {
+            return 0L
+        }
+
+        val memorySizeKb = memInfoFile.readLinesSafe(internalLogger = internalLogger)?.firstNotNullOfOrNull { line ->
```

Comment:
minor: maybe it is easier to use: `filterNotNull + firstOrNull` instead? I find `firstNotNullOfOrNull` hard to read/understand

Reply:
break down to `mapNotNull` + `.firstOrNull`